app/controllers/verify/step_controller.rb

```diff
@@ -8,5 +8,25 @@ class StepController < ApplicationController
     before_action :confirm_idv_attempts_allowed
 
     helper_method :step
+
+    private
+
+    def step_name
+      raise NotImplementedError, 'must implement step_name method'
+    end
+
+    def increment_step_attempts(increments = 1)
+      idv_session.step_attempts[step_name] += increments
+    end
+
+    def step_attempts_exceeded?
+      idv_session.step_attempts[step_name] >= Idv::Attempter.idv_max_attempts
+    end
+
+    def confirm_step_allowed
```

Comment:
I'm not sure that this name makes sense....but I don't have a good alternative off-hand? `show_fail_after_max_attempts` ?